packages/react/src/components/DetailsList/DetailsRow.base.tsx

```diff
@@ -283,10 +283,16 @@ export class DetailsRowBase extends React.Component<IDetailsRowBaseProps, IDetai
 
     // When the user does not specify any column is a row-header in the columns props,
     // The aria-labelledby of the checkbox does not specify {id}-header.
-    const isColumnsHasHeader = columns.some(column => {
-      return !!column.isRowHeader;
-    });
-    const ariaLabelledby = `${id}-checkbox` + (isColumnsHasHeader ? ` ${id}-header` : '');
+    let ariaLabelledby: string | undefined;
+    if (id) {
+      ariaLabelledby = `${id}-checkbox`;
+      const hasRowHeader = columns.some(column => {
```

Comment:
I might be wrong but doesn't the `hasRowHeader...` part need to be outside of the `if (id)...` block? In other words, isn't it possible that `hasRowHeader` might be `true` when `id` is falsey?

Reply:
@spmonahan thank you for your reply.
Modified here, because I refer to the [original logic](https://github.com/microsoft/fluentui/pull/21549/files#diff-79072ab4cd5f5ec36341419915356ba9e623775de7f697b1392727f354718e36L329).

```typescript
'aria-labelledby': id ? `${id}-checkbox ${id}-header` : undefined,
```

when ```id``` is false, ```hasRowHeader``` doesn't make sense for whatever cases.
To minimize the impact of this change, I recommend adopting the logic from the [last commit](https://github.com/microsoft/fluentui/pull/21549/commits/50addf0c21aa2127272334270e0c54b2b6991845).

If you agree to do so, I will commit to rollback to the last time.